### PR TITLE
HPCC-26704 Fix Coverity scan defects related to soapplus

### PR DESCRIPTION
**Pull request by wangkx**
1. Control flow issue in expandConciseRequest().
2. Calling "read" without checking return value in Http::receiveData().
3. Null pointer dereferences for m_globals and globals.
4. Error handling issue for m_sockfd.
5. Calling "inet_pton" without checking return value in CAddress::CAddress().
6. In MessageGenerator::generateMessage(), assigning the return value of
"getchar" to char "c" truncates its value.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/tools/soapplus/msggenerator.cpp [hunk 1]
```diff
@@ -43,25 +43,24 @@ static bool loadFile(StringBuffer& s, const char* file)
...
-    m_gx = globals->queryProp("gx") ? true : false;
-    m_soapWrap = globals->queryProp("gs") ? false: true;
-    m_ecl2esp = globals->getPropBool("ECL2ESP", false);
+    m_globals.setown(globals ? LINK(globals) : createProperties(true));
+
+    m_items = m_globals->getPropInt("items", 1);
+    m_isRoxie = m_globals->queryProp("roxie") ? true : false;
+    m_genAllDatasets = m_globals->queryProp("alldataset") ? true : false;
+    m_gx = m_globals->queryProp("gx") ? true : false;
+    m_soapWrap = m_globals->queryProp("gs") ? false: true;
```
Comment (afishbeck) [inline esp/tools/soapplus/msggenerator.cpp:52]: Space before colon?

**Review by afishbeck — APPROVED** [review]
@wangkx looks good, except one trivial comment.

File: esp/tools/soapplus/msggenerator.cpp (hunk 1 above)
Comment (wangkx) [reply esp/tools/soapplus/msggenerator.cpp]: I just added a space.

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged

**Pull request merged**